### PR TITLE
deepstream: update to 6.0

### PR DESCRIPTION
**Pull request by ichergui**
from JetPack 4.6 (rev 2)

Signed-off-by: Ilies CHERGUI <ilies.chergui@gmail.com>

### CODE REVIEW

**Comment by ichergui** [comment]
* This PR is here to address issue #837
* tests were done by using `jetson-tx2` 
```
# deepstream-app -c /opt/nvidia/deepstream/deepstream-6.0/samples/configs/deepstream-app/source12_1080p_dec_infer-resnet_tracker_tiled_display_fp16_tx2.txt
```

* Result 

```
gstnvtracker: Loading low-level lib at /opt/nvidia/deepstream/deepstream-6.0/lib/libnvds_nvmultiobjecttracker.so
gstnvtracker: Batch processing is ON
gstnvtracker: Past frame output is ON
[NvMultiObjectTracker] Initialized
0:00:03.996191750  5623      0x2541130 INFO                 nvinfer gstnvinfer.cpp:638:gst_nvinfer_logger:<primary_gie> NvDsInferContext[UID 1]: Info from NvDsInferContextImpl::deserializeEngineAndBackend(
) <nvdsinfer_context_impl.cpp:1900> [UID = 1]: deserialized trt engine from :/opt/nvidia/deepstream/deepstream-6.0/samples/configs/deepstream-app/../../models/Primary_Detector_Nano/resnet10.caffemodel_b12_
gpu0_fp16.engine
0:00:03.996461828  5623      0x2541130 INFO                 nvinfer gstnvinfer.cpp:638:gst_nvinfer_logger:<primary_gie> NvDsInferContext[UID 1]: Info from NvDsInferContextImpl::generateBackendContext() <nv
dsinfer_context_impl.cpp:2004> [UID = 1]: Use deserialized engine model: /opt/nvidia/deepstream/deepstream-6.0/samples/configs/deepstream-app/../../models/Primary_Detector_Nano/resnet10.caffemodel_b12_gpu0
_fp16.engine
0:00:04.013756142  5623      0x2541130 INFO                 nvinfer gstnvinfer_impl.cpp:313:notifyLoadModelStatus:<primary_gie> [UID 1]: Load new model:/opt/nvidia/deepstream/deepstream-6.0/samples/configs
/deepstream-app/config_infer_primary_nano.txt sucessfully
INFO: [Implicit Engine Info]: layers num: 3
0   INPUT  kFLOAT input_1         3x272x480       
1   OUTPUT kFLOAT conv2d_bbox     16x17x30 

...

Debug info: ../../../../gstreamer-1.14.4/libs/gst/base/gstbasesink.c(2902): gst_base_sink_is_too_late (): /GstPipeline:pipeline/GstBin:processing_bin_0/GstBin:sink_bin/GstBin:sink_sub_bin1/GstNvOverlaySink
-nvoverlaysink:sink_sub_bin_sink1:
There may be a timestamping problem, or this computer is too slow.
WARNING from sink_sub_bin_sink1: A lot of buffers are being dropped.
Debug info: ../../../../gstreamer-1.14.4/libs/gst/base/gstbasesink.c(2902): gst_base_sink_is_too_late (): /GstPipeline:pipeline/GstBin:processing_bin_0/GstBin:sink_bin/GstBin:sink_sub_bin1/GstNvOverlaySink
-nvoverlaysink:sink_sub_bin_sink1:
There may be a timestamping problem, or this computer is too slow.
** INFO: <bus_callback:217>: Received EOS. Exiting ...

Quitting
[NvMultiObjectTracker] De-initialized
App run successful
root@jetson-tx2-devkit:~#
```

File: recipes-devtools/deepstream/deepstream-6.0_6.0.0-1.bb [hunk 1]
```diff
@@ -104,7 +97,6 @@ FILES:${PN}-samples-data = "\
 FILES:${PN}-sources = "${DEEPSTREAM_PATH}/sources"
 
 FILES:${PN}-iothub-client = "${DEEPSTREAM_PATH}/lib/libiothub_client.so"
-FILES:${PN}-azure-edge = "${DEEPSTREAM_PATH}/lib/libazure*"
```
Comment (madisongh) [inline recipes-devtools/deepstream/deepstream-6.0_6.0.0-1.bb:107]: I think we still need the `-azure-edge` package, but the libraries to put there are `libnvds_azure*`.

File: recipes-devtools/deepstream/deepstream-6.0_6.0.0-1.bb [hunk 2]
```diff
@@ -104,7 +97,6 @@ FILES:${PN}-samples-data = "\
...
 FILES:${PN}-triton-plugins = "${DEEPSTREAM_PATH}/lib/triton_backends"
```
Comment (madisongh) [inline recipes-devtools/deepstream/deepstream-6.0_6.0.0-1.bb:100]: I don't think this directory exists any longer, so maybe we can drop this package.

I'm a bit concerned, though, that `libnvds_infer_server` and the `libnvdsgst_inferserver` gstreamer plugin both depend on Triton, and was thinking that perhaps we should package those separately.

File: recipes-devtools/deepstream/deepstream-6.0_6.0.0-1.bb [hunk 3]
```diff
@@ -47,14 +48,6 @@ do_install() {
         install -m 0644 "$f" ${D}${DEEPSTREAM_PATH}/lib/
     done
     ln -sf libnvds_msgconv.so.1.0.0 ${D}${DEEPSTREAM_PATH}/lib/libnvds_msgconv.so
```
Comment (madisongh) [inline recipes-devtools/deepstream/deepstream-6.0_6.0.0-1.bb:50]: There's also a new `libnvds_msgconv_audio` library, and we might need a similar symlink for that one.

Also, I don't see you installing the libraries under `lib/cvcore_libs`.

**Review by madisongh — CHANGES_REQUESTED** [review]

File: recipes-devtools/deepstream/deepstream-6.0_6.0.0-1.bb (hunk 1 above)
Comment (ichergui) [reply recipes-devtools/deepstream/deepstream-6.0_6.0.0-1.bb:107]: Yes, you're right.
I will update the recipe

File: recipes-devtools/deepstream/deepstream-6.0_6.0.0-1.bb (hunk 3 above)
Comment (ichergui) [reply recipes-devtools/deepstream/deepstream-6.0_6.0.0-1.bb:50]: I was not sure if we need it. 
I will update the recipe

File: recipes-devtools/deepstream/deepstream-6.0_6.0.0-1.bb [hunk 4]
```diff
@@ -47,14 +48,9 @@ do_install() {
...
-    # Requires Python 3.6
-    rm -f ${D}${DEEPSTREAM_PATH}/lib/pyds.so
+    ln -sf libnvds_msgconv_audio.so.1.0.0 ${D}${DEEPSTREAM_PATH}/lib/libnvds_msgconv_audio.so
 
-    for i in 1 2; do
-        path="${DEEPSTREAM_PATH}/lib/triton_backends/tensorflow$i"
-	install -d ${D}$path
-	cp --preserve=mode,timestamps,links --no-dereference ${S}$path/* ${D}$path/
-    done
+    install -m 0644 ${S}${DEEPSTREAM_PATH}/lib/cvcore_libs/* ${D}${DEEPSTREAM_PATH}/lib/
```
Comment (madisongh) [inline recipes-devtools/deepstream/deepstream-6.0_6.0.0-1.bb:53]: The install target should also be ...`lib/cvcore_libs/` to match the original structure

**Review by madisongh — CHANGES_REQUESTED** [review]

File: recipes-devtools/deepstream/deepstream-6.0_6.0.0-1.bb (hunk 4 above)
Comment (ichergui) [reply recipes-devtools/deepstream/deepstream-6.0_6.0.0-1.bb]: Done

File: recipes-devtools/deepstream/deepstream-6.0_6.0.0-1.bb (hunk 2 above)
Comment (ichergui) [reply recipes-devtools/deepstream/deepstream-6.0_6.0.0-1.bb]: Please check the new implementation

**Comment by ichergui** [comment]
* Tests on `Jetson 2`

[running_deepstream-app.log](https://github.com/OE4T/meta-tegra/files/7516889/running_deepstream-app.log)
[running_deepstream-test1-app.log](https://github.com/OE4T/meta-tegra/files/7516890/running_deepstream-test1-app.log)
[running_deepstream-test2-app.log](https://github.com/OE4T/meta-tegra/files/7516891/running_deepstream-test2-app.log)
[running_deepstream-test3-app_part1.log](https://github.com/OE4T/meta-tegra/files/7516983/running_deepstream-test3-app_part1.log)

**Review by madisongh — APPROVED** [review]

**Pull request merged**